test_submission_schema/schemas/v1.json

```diff
@@ -0,0 +1,177 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "$id": "https://github.com/canonical/test-submission-schema/blob/main/test_submission_schema/schemas/v1.json",
+    "title": "Test Submission Schema",
+    "description": "A submission schema for submitting test plans.",
+    "type": "object",
+    "additionalProperties": false,
+    "required": [
+        "version",
+        "origin",
+        "session_data",
+        "results"
+    ],
+    "properties": {
+        "version": {
+            "const": 1,
+            "description": "The version of the test submission schema."
+        },
+        "origin": {
+            "type": "object",
+            "required": [
+                "name",
+                "version",
+                "packaging"
+            ],
+            "properties": {
+                "name": {
+                    "type": "string",
+                    "description": "The application used to run the test plan."
+                },
+                "version": {
+                    "type": "string",
+                    "description": "The version of the application used to run the test plan."
+                },
+                "packaging": {
+                    "type": "object",
+                    "properties": {
+                        "type": {
+                            "type": "string",
+                            "description": "The type of application. e.g. Snap, Deb."
+                        },
+                        "name": {
+                            "type": "string",
+                            "description": "The name of the application."
+                        },
+                        "version": {
+                            "type": "string",
+                            "description": "The version of the application."
+                        },
+                        "revision": {
+                            "type": "string",
+                            "description": "The revision of the application."
+                        },
+                        "date": {
+                            "type": "string",
+                            "format": "date",
+                            "description": "The package creation date."
+                        }
+                    },
+                    "if": {"properties": {"type": {"const": "snap"}}},
+                    "then": {"required": ["type", "name", "version", "revision", "date"]},
+                    "else": {"required": ["type", "name", "version", "date"]}
```

Comment:
If `date` is removed, it should be removed from here too.

Reply:
also, if the type is `source`, `name` is not required (there is no "installed package" per se).